## Compare GINCO and MT-GINCO labels with FTD

Based on the GINCO and MT-GINCO predictions that we applied to the FTD, I'll analyse how the FTD labels are connected with GINCO labels.

In [1]:
import pandas as pd
import numpy as np

In [11]:
# Import the FTD dataset
ftd = pd.read_csv("data-sheets-with-all-info/FTD-dataset-with-all-information.csv", sep="\t", index_col=0)
ftd.head(3)

ID              labels Multiple labels  \
0  __id__1-syndicate  A1 (argumentative)             NaN   
1  __id__2-syndicate  A1 (argumentative)             NaN   
2  __id__3-syndicate  A1 (argumentative)             NaN   

                                                text  length  \
0  BMW's and Chinese Justice * * * * * In most pl...     975   
1  China and a New Balance of Power SHANGHAI – Th...     956   
2  China and Russia in the New World Disorder Can...     978   

       GINCO_downcast_pred MT-GINCO_downcast_pred  
0           News/Reporting         News/Reporting  
1  Information/Explanation  Opinion/Argumentation  
2    Opinion/Argumentation  Opinion/Argumentation

In [12]:
# Analyse agreement of the labels
ftd["agreement"] = np.where(ftd['GINCO_downcast_pred'] == ftd['MT-GINCO_downcast_pred'], "yes", "no")

ftd["agreement"].value_counts()

yes    1206
no      347
Name: agreement, dtype: int64

In [13]:
347/1206

0.2877280265339967

The predictions of GINCO and MT-GINCO classifier differ in 347 instances (29%).

In [15]:
ftd.columns

Index(['ID', 'labels', 'Multiple labels', 'text', 'length',
       'GINCO_downcast_pred', 'MT-GINCO_downcast_pred', 'agreement'],
      dtype='object')

In [16]:
ftd["Multiple labels"].value_counts()

y    139
Name: Multiple labels, dtype: int64

In [18]:
# Discard texts with multiple labels
ftd = ftd[ftd["Multiple labels"] != "y"]
ftd.describe(include="all")

ID              labels Multiple labels  \
count                1414                1414               0   
unique               1414                  10               0   
top     __id__1-syndicate  A1 (argumentative)             NaN   
freq                    1                 296             NaN   
mean                  NaN                 NaN             NaN   
std                   NaN                 NaN             NaN   
min                   NaN                 NaN             NaN   
25%                   NaN                 NaN             NaN   
50%                   NaN                 NaN             NaN   
75%                   NaN                 NaN             NaN   
max                   NaN                 NaN             NaN   

                                                     text         length  \
count                                                1414    1414.000000   
unique                                               1414            NaN   
top     BMW's and Chinese Justice * * * * * In most pl...            NaN   
freq                                                    1            NaN   
mean                                                  NaN    1445.212164   
std                                                   NaN    4989.570842   
min                                                   NaN      31.000000   
25%                                                   NaN     224.000000   
50%                                                   NaN     495.000000   
75%                                                   NaN    1144.250000   
max                                                   NaN  146922.000000   

            GINCO_downcast_pred   MT-GINCO_downcast_pred agreement  
count                      1414                     1414      1414  
unique                        9                        9         2  
top     Information/Explanation  Information/Explanation       yes  
freq                        407                      311      1118  
mean                        NaN                      NaN       NaN  
std                         NaN                      NaN       NaN  
min                         NaN                      NaN       NaN  
25%                         NaN                      NaN       NaN  
50%                         NaN                      NaN       NaN  
75%                         NaN                      NaN       NaN  
max                         NaN                      NaN       NaN

In [19]:
# Analyse how GINCO primary labels are connected with FTD labels based on the prediction
pd.crosstab(ftd['labels'], ftd["GINCO_downcast_pred"], normalize="index")


GINCO_downcast_pred     Forum  Information/Explanation  Instruction  \
labels                                                                
A1 (argumentative)   0.013514                 0.263514     0.000000   
A11 (personal)       0.088608                 0.037975     0.012658   
A12 (promotion)      0.011719                 0.156250     0.023438   
A14 (academic)       0.000000                 0.810127     0.000000   
A16 (information)    0.000000                 0.815476     0.000000   
A17 (review)         0.088235                 0.191176     0.000000   
A4 (fiction)         0.000000                 0.202128     0.000000   
A7 (instruction)     0.000000                 0.139394     0.612121   
A8 (news)            0.000000                 0.095588     0.000000   
A9 (legal)           0.000000                 0.232877     0.013699   

GINCO_downcast_pred  Legal/Regulation  List of Summaries/Excerpts  \
labels                                                              
A1 (argumentative)           0.010135                    0.091216   
A11 (personal)               0.000000                    0.037975   
A12 (promotion)              0.000000                    0.070312   
A14 (academic)               0.012658                    0.012658   
A16 (information)            0.005952                    0.053571   
A17 (review)                 0.000000                    0.147059   
A4 (fiction)                 0.000000                    0.000000   
A7 (instruction)             0.000000                    0.090909   
A8 (news)                    0.014706                    0.110294   
A9 (legal)                   0.643836                    0.000000   

GINCO_downcast_pred  News/Reporting  Opinion/Argumentation     Other  \
labels                                                                 
A1 (argumentative)         0.246622               0.229730  0.094595   
A11 (personal)             0.025316               0.696203  0.101266   
A12 (promotion)            0.093750               0.015625  0.035156   
A14 (academic)             0.037975               0.088608  0.037975   
A16 (information)          0.065476               0.023810  0.005952   
A17 (review)               0.132353               0.132353  0.014706   
A4 (fiction)               0.021277               0.234043  0.542553   
A7 (instruction)           0.030303               0.030303  0.036364   
A8 (news)                  0.742647               0.007353  0.029412   
A9 (legal)                 0.013699               0.000000  0.095890   

GINCO_downcast_pred  Promotion  
labels                          
A1 (argumentative)    0.050676  
A11 (personal)        0.000000  
A12 (promotion)       0.593750  
A14 (academic)        0.000000  
A16 (information)     0.029762  
A17 (review)          0.294118  
A4 (fiction)          0.000000  
A7 (instruction)      0.060606  
A8 (news)             0.000000  
A9 (legal)            0.000000

In [20]:
pd.crosstab(ftd['labels'], ftd["GINCO_downcast_pred"], normalize="index").to_dict("index")

{'A1 (argumentative)': {'Forum': 0.013513513513513514,
  'Information/Explanation': 0.2635135135135135,
  'Instruction': 0.0,
  'Legal/Regulation': 0.010135135135135136,
  'List of Summaries/Excerpts': 0.09121621621621621,
  'News/Reporting': 0.24662162162162163,
  'Opinion/Argumentation': 0.22972972972972974,
  'Other': 0.0945945945945946,
  'Promotion': 0.05067567567567568},
 'A11 (personal)': {'Forum': 0.08860759493670886,
  'Information/Explanation': 0.0379746835443038,
  'Instruction': 0.012658227848101266,
  'Legal/Regulation': 0.0,
  'List of Summaries/Excerpts': 0.0379746835443038,
  'News/Reporting': 0.02531645569620253,
  'Opinion/Argumentation': 0.6962025316455697,
  'Other': 0.10126582278481013,
  'Promotion': 0.0},
 'A12 (promotion)': {'Forum': 0.01171875,
  'Information/Explanation': 0.15625,
  'Instruction': 0.0234375,
  'Legal/Regulation': 0.0,
  'List of Summaries/Excerpts': 0.0703125,
  'News/Reporting': 0.09375,
  'Opinion/Argumentation': 0.015625,
  'Other': 0.03515

What we can see based on the GINCO predictions on the FTD labels, is (first, the information for prediction of GINCO-downcast is given, followed by the information what is different on predictions by the MT-GINCO-downcast model):)


## Compare GINCO labels with CORE

In [3]:
# Import the CORE dataset
core_df = pd.read_csv("data-sheets-with-all-info/CORE-all-information.csv", index_col = 0, sep="\t")

core_df.head(3)

label                                               text  split  \
0  NA OP SR OB  The Top TEN 'Whiniest Sets of Fans' in English...  train   
1        NA NE  Ferry consultation needs deeper questions, say...  train   
2        ID DF  I'v been recording and mixing music for about ...  train   

  main_labels sublabels  Len  main_len  sub_len      GINCORE  \
0       NA OP     SR OB    4         2        2  NA OP SR OB   
1         NaN        NE    2         1        1         News   
2          ID        DF    2         1        1        Forum   

         full_names  main_labels_full_names FTD_pred GINCO_downcast_pred  \
0             SR OB                   NA OP      NaN                 NaN   
1  News Report/Blog               Narrative      NaN                 NaN   
2  Discussion Forum  Interactive Discussion      NaN                 NaN   

  MT-GINCO_downcast_pred  
0                    NaN  
1                    NaN  
2                    NaN

In [4]:
# Filter out only instances that have GINCO prediction
core_df = core_df.dropna(subset = ["GINCO_downcast_pred"])

core_df.shape

(1500, 14)

In [5]:
core_df.columns

Index(['label', 'text', 'split', 'main_labels', 'sublabels', 'Len', 'main_len',
       'sub_len', 'GINCORE', 'full_names', 'main_labels_full_names',
       'FTD_pred', 'GINCO_downcast_pred', 'MT-GINCO_downcast_pred'],
      dtype='object')

In [6]:
# Discard uninteresting columns
core_df = core_df[['GINCORE', 'sub_len', 'full_names', 'main_len', 'main_labels_full_names','FTD_pred', 'GINCO_downcast_pred', 'MT-GINCO_downcast_pred']]

core_df

GINCORE  sub_len        full_names  main_len  \
14                      News        1     Sports Report         1   
16                      News        1  News Report/Blog         1   
23                      News        1  News Report/Blog         1   
54                      News        1  News Report/Blog         1   
57                      News        1  News Report/Blog         1   
...                      ...      ...               ...       ...   
48378  Opinion/Argumentation        1      Opinion Blog         1   
48403                   News        1  News Report/Blog         1   
48405  Opinion/Argumentation        1      Opinion Blog         1   
48419                   News        1  News Report/Blog         1   
48423                   News        1  News Report/Blog         1   

      main_labels_full_names            FTD_pred         GINCO_downcast_pred  \
14                 Narrative  A1 (argumentative)       Opinion/Argumentation   
16                 Narrative           A8 (news)              News/Reporting   
23                 Narrative  A1 (argumentative)       Opinion/Argumentation   
54                 Narrative        A17 (review)              News/Reporting   
57                 Narrative        A17 (review)  List of Summaries/Excerpts   
...                      ...                 ...                         ...   
48378                Opinion  A1 (argumentative)     Information/Explanation   
48403              Narrative  A1 (argumentative)              News/Reporting   
48405                Opinion  A1 (argumentative)              News/Reporting   
48419              Narrative           A8 (news)              News/Reporting   
48423              Narrative           A8 (news)              News/Reporting   

           MT-GINCO_downcast_pred  
14          Opinion/Argumentation  
16                 News/Reporting  
23          Opinion/Argumentation  
54                 News/Reporting  
57     List of Summaries/Excerpts  
...                           ...  
48378     Information/Explanation  
48403              News/Reporting  
48405              News/Reporting  
48419              News/Reporting  
48423              News/Reporting  

[1500 rows x 8 columns]

In [7]:
# Analyze in how many instance do GINCO and MT-GINCO predictions differ

core_df["agreement"] = np.where(core_df['GINCO_downcast_pred'] == core_df['MT-GINCO_downcast_pred'], "yes", "no")

core_df

GINCORE  sub_len        full_names  main_len  \
14                      News        1     Sports Report         1   
16                      News        1  News Report/Blog         1   
23                      News        1  News Report/Blog         1   
54                      News        1  News Report/Blog         1   
57                      News        1  News Report/Blog         1   
...                      ...      ...               ...       ...   
48378  Opinion/Argumentation        1      Opinion Blog         1   
48403                   News        1  News Report/Blog         1   
48405  Opinion/Argumentation        1      Opinion Blog         1   
48419                   News        1  News Report/Blog         1   
48423                   News        1  News Report/Blog         1   

      main_labels_full_names            FTD_pred         GINCO_downcast_pred  \
14                 Narrative  A1 (argumentative)       Opinion/Argumentation   
16                 Narrative           A8 (news)              News/Reporting   
23                 Narrative  A1 (argumentative)       Opinion/Argumentation   
54                 Narrative        A17 (review)              News/Reporting   
57                 Narrative        A17 (review)  List of Summaries/Excerpts   
...                      ...                 ...                         ...   
48378                Opinion  A1 (argumentative)     Information/Explanation   
48403              Narrative  A1 (argumentative)              News/Reporting   
48405                Opinion  A1 (argumentative)              News/Reporting   
48419              Narrative           A8 (news)              News/Reporting   
48423              Narrative           A8 (news)              News/Reporting   

           MT-GINCO_downcast_pred agreement  
14          Opinion/Argumentation       yes  
16                 News/Reporting       yes  
23          Opinion/Argumentation       yes  
54                 News/Reporting       yes  
57     List of Summaries/Excerpts       yes  
...                           ...       ...  
48378     Information/Explanation       yes  
48403              News/Reporting       yes  
48405              News/Reporting       yes  
48419              News/Reporting       yes  
48423              News/Reporting       yes  

[1500 rows x 9 columns]

In [8]:
core_df.agreement.value_counts()

yes    1235
no      265
Name: agreement, dtype: int64

In [9]:
265/1500

0.17666666666666667

The GINCO and MT-GINCO predictions differ only in case of 265 instances (18% of instances).

### CORE-main

In [6]:
# Analyse how CORE main labels are connected with GINCO labels based on the prediction

# Filter out instances that have multiple main labels
main_labels = core_df[core_df["main_len"] == 1]

pd.crosstab(main_labels["main_labels_full_names"], main_labels["FTD_pred"], normalize="index")

FTD_pred                               A1 (argumentative)  A11 (personal)  \
main_labels_full_names                                                      
How-To/Instructional                             0.043011        0.118280   
Informational Description/Explanation            0.189058        0.020776   
Informational Persuasion                         0.185345        0.021552   
Interactive Discussion                           0.315702        0.289256   
Lyrical                                          0.040650        0.577236   
Narrative                                        0.228284        0.133191   
Opinion                                          0.467799        0.101203   
Spoken                                           0.304348        0.278261   

FTD_pred                               A12 (promotion)  A14 (academic)  \
main_labels_full_names                                                   
How-To/Instructional                          0.093190        0.000000   
Informational Description/Explanation         0.228532        0.052632   
Informational Persuasion                      0.400862        0.000000   
Interactive Discussion                        0.047934        0.009917   
Lyrical                                       0.048780        0.000000   
Narrative                                     0.032002        0.002743   
Opinion                                       0.061571        0.000708   
Spoken                                        0.060870        0.000000   

FTD_pred                               A16 (information)  A17 (review)  \
main_labels_full_names                                                   
How-To/Instructional                            0.003584      0.021505   
Informational Description/Explanation           0.213989      0.089335   
Informational Persuasion                        0.030172      0.258621   
Interactive Discussion                          0.028099      0.041322   
Lyrical                                         0.032520      0.121951   
Narrative                                       0.029259      0.067662   
Opinion                                         0.012739      0.230715   
Spoken                                          0.000000      0.252174   

FTD_pred                               A4 (fiction)  A7 (instruction)  \
main_labels_full_names                                                  
How-To/Instructional                       0.000000          0.713262   
Informational Description/Explanation      0.004848          0.106648   
Informational Persuasion                   0.012931          0.077586   
Interactive Discussion                     0.008264          0.239669   
Lyrical                                    0.089431          0.081301   
Narrative                                  0.017373          0.006400   
Opinion                                    0.012739          0.083510   
Spoken                                     0.017391          0.008696   

FTD_pred                               A8 (news)  A9 (legal)  
main_labels_full_names                                        
How-To/Instructional                    0.000000    0.007168  
Informational Description/Explanation   0.059557    0.034626  
Informational Persuasion                0.004310    0.008621  
Interactive Discussion                  0.014876    0.004959  
Lyrical                                 0.008130    0.000000  
Narrative                               0.480951    0.002133  
Opinion                                 0.029016    0.000000  
Spoken                                  0.078261    0.000000

In [10]:
main_labels["main_labels_full_names"].value_counts()

Narrative                                3281
Informational Description/Explanation    1444
Opinion                                  1413
Interactive Discussion                    605
How-To/Instructional                      279
Informational Persuasion                  232
Lyrical                                   123
Spoken                                    115
Name: main_labels_full_names, dtype: int64

In [11]:
main_labels["main_labels_full_names"].unique()

array(['Interactive Discussion', 'Narrative', 'Opinion',
       'Informational Description/Explanation', 'How-To/Instructional',
       'Lyrical', 'Informational Persuasion', 'Spoken'], dtype=object)

In [12]:
pd.crosstab(main_labels["main_labels_full_names"], main_labels["FTD_pred"], normalize="index").to_dict("index")

{'How-To/Instructional': {'A1 (argumentative)': 0.043010752688172046,
  'A11 (personal)': 0.11827956989247312,
  'A12 (promotion)': 0.0931899641577061,
  'A14 (academic)': 0.0,
  'A16 (information)': 0.0035842293906810036,
  'A17 (review)': 0.021505376344086023,
  'A4 (fiction)': 0.0,
  'A7 (instruction)': 0.7132616487455197,
  'A8 (news)': 0.0,
  'A9 (legal)': 0.007168458781362007},
 'Informational Description/Explanation': {'A1 (argumentative)': 0.18905817174515235,
  'A11 (personal)': 0.02077562326869806,
  'A12 (promotion)': 0.22853185595567868,
  'A14 (academic)': 0.05263157894736842,
  'A16 (information)': 0.21398891966759004,
  'A17 (review)': 0.08933518005540166,
  'A4 (fiction)': 0.004847645429362881,
  'A7 (instruction)': 0.10664819944598337,
  'A8 (news)': 0.05955678670360111,
  'A9 (legal)': 0.03462603878116344},
 'Informational Persuasion': {'A1 (argumentative)': 0.1853448275862069,
  'A11 (personal)': 0.021551724137931036,
  'A12 (promotion)': 0.40086206896551724,
  'A14 

Comparison of main CORE labels and FTD labels:

1. Well-connected:
* 'How-To/Instructional': 'A7 (instruction)': 0.713

2. Not well connected (no clear majority label/majority label does not seem to be appropriate):
* 'Interactive Discussion': mostly 'A1 (argumentative)': 0.315 (percentage of instances of Interactive Discussion, identified as A1) + 'A11 (personal)': 0.289,  'A7 (instruction)': 0.239
* 'Narrative': 'A8 (news)': 0.48, 'A1 (argumentative)': 0.228
* 'Opinion': 'A1 (argumentative)': 0.467, 'A17 (review)': 0.230
* 'Informational Description/Explanation': A12 (promotion)': 0.228, 'A16 (information)': 0.21, 'A1 (argumentative)': 0.189
* 'Lyrical': 'A11 (personal)': 0.577
* 'Informational Persuasion': 'A12 (promotion)': 0.40, 'A1 (argumentative)': 0.185, 'A17 (review)': 0.258
* 'Spoken': 'A1 (argumentative)': 0.30, 'A11 (personal)': 0.278, 'A17 (review)': 0.252




### CORE-sub

In [13]:
core_df.head(2)

GINCORE  sub_len        full_names  main_len  main_labels_full_names  \
6    Forum        1  Discussion Forum         1  Interactive Discussion   
14    News        1     Sports Report         1               Narrative   

              FTD_pred  
6       A11 (personal)  
14  A1 (argumentative)

In [14]:
# Analyse how CORE sub labels are connected with FTD labels based on the prediction

# Filter out instances that have multiple main labels
sub_labels = core_df[core_df["sub_len"] == 1]

pd.crosstab(sub_labels["full_names"], sub_labels["FTD_pred"], normalize="index")

FTD_pred                         A1 (argumentative)  A11 (personal)  \
full_names                                                            
Advertisement                              0.000000        0.000000   
Advice                                     0.172043        0.139785   
Course Materials                           0.173913        0.000000   
Description of a Person                    0.098039        0.058824   
Description of a Thing                     0.139800        0.021398   
Description with Intent to Sell            0.082192        0.018265   
Discussion Forum                           0.374359        0.320513   
Editorial                                  0.923077        0.000000   
Encyclopedia Article                       0.095238        0.009524   
FAQ about How-to                           0.000000        0.000000   
FAQ about Information                      0.060000        0.020000   
Formal Speech                              0.777778        0.111111   
Historical Article                         0.364706        0.000000   
How-to                                     0.057034        0.076046   
Information Blog                           0.290850        0.039216   
Interview                                  0.180851        0.265957   
Legal terms                                0.052632        0.000000   
Letter to Editor                           0.875000        0.125000   
Magazine Article                           0.466667        0.200000   
News Report/Blog                           0.284626        0.016659   
Opinion Blog                               0.660218        0.123337   
Other Forum                                0.000000        0.750000   
Other Information                          0.000000        0.074074   
Other Informational Persuasion             1.000000        0.000000   
Other Opinion                              0.666667        0.000000   
Other Spoken                               0.333333        0.666667   
Personal Blog                              0.122744        0.676895   
Persuasive Article or Essay                0.750000        0.041667   
Poem                                       0.066667        0.200000   
Prayer                                     0.000000        0.000000   
Question/Answer Forum                      0.209524        0.233333   
Reader/Viewer Responses                    0.100000        0.100000   
Recipe                                     0.000000        0.352941   
Religious Blogs/Sermons                    0.671429        0.042857   
Research Article                           0.387879        0.006061   
Reviews                                    0.077778        0.066667   
Short Story                                0.017857        0.160714   
Song Lyrics                                0.037037        0.629630   
Sports Report                              0.163121        0.023050   
TV/Movie Script                            0.000000        0.600000   
Technical Report                           0.375000        0.125000   
Technical Support                          0.000000        0.125000   
Transcript of Video/Audio                  0.833333        0.083333   
Travel Blog                                0.070175        0.438596   

FTD_pred                         A12 (promotion)  A14 (academic)  \
full_names                                                         
Advertisement                           0.666667        0.000000   
Advice                                  0.145161        0.000000   
Course Materials                        0.130435        0.043478   
Description of a Person                 0.143791        0.000000   
Description of a Thing                  0.370899        0.007133   
Description with Intent to Sell         0.465753        0.000000   
Discussion Forum                        0.041026        0.002564   
Editorial                               0.000000        0.000000   
Encyclopedia Article                    0.000000     

In [15]:
sub_labels["full_names"].value_counts()

News Report/Blog                   2101
Opinion Blog                        827
Description of a Thing              701
Sports Report                       564
Personal Blog                       554
Discussion Forum                    390
Reviews                             360
Information Blog                    306
How-to                              263
Description with Intent to Sell     219
Question/Answer Forum               210
Advice                              186
Research Article                    165
Description of a Person             153
Religious Blogs/Sermons             140
Song Lyrics                         108
Encyclopedia Article                105
Interview                            94
Historical Article                   85
Travel Blog                          57
Short Story                          56
FAQ about Information                50
Legal terms                          38
Recipe                               34
Other Information                    27


In [16]:
len(sub_labels.full_names.unique())

44

In [18]:
pd.crosstab(sub_labels["full_names"], sub_labels["FTD_pred"], normalize="index").to_dict("index")

{'Advertisement': {'A1 (argumentative)': 0.0,
  'A11 (personal)': 0.0,
  'A12 (promotion)': 0.6666666666666666,
  'A14 (academic)': 0.0,
  'A16 (information)': 0.0,
  'A17 (review)': 0.0,
  'A4 (fiction)': 0.0,
  'A7 (instruction)': 0.3333333333333333,
  'A8 (news)': 0.0,
  'A9 (legal)': 0.0},
 'Advice': {'A1 (argumentative)': 0.17204301075268819,
  'A11 (personal)': 0.13978494623655913,
  'A12 (promotion)': 0.14516129032258066,
  'A14 (academic)': 0.0,
  'A16 (information)': 0.005376344086021506,
  'A17 (review)': 0.016129032258064516,
  'A4 (fiction)': 0.0,
  'A7 (instruction)': 0.5161290322580645,
  'A8 (news)': 0.005376344086021506,
  'A9 (legal)': 0.0},
 'Course Materials': {'A1 (argumentative)': 0.17391304347826086,
  'A11 (personal)': 0.0,
  'A12 (promotion)': 0.13043478260869565,
  'A14 (academic)': 0.043478260869565216,
  'A16 (information)': 0.30434782608695654,
  'A17 (review)': 0.043478260869565216,
  'A4 (fiction)': 0.0,
  'A7 (instruction)': 0.30434782608695654,
  'A8 (ne

1. Categories that match well:
* 'Advertisement': 'A12 (promotion)': 0.67
* 'Editorial': 'A1 (argumentative)': 0.92
* 'Encyclopedia Article': 'A16 (information)': 0.82
* 'FAQ about How-to':  'A7 (instruction)': 0.66
* 'How-to':  'A7 (instruction)': 0.74
* 'Formal Speech': 'A1 (argumentative)': 0.78
* 'Legal terms': 'A9 (legal)': 0.789
* 'Letter to Editor': 'A1 (argumentative)': 0.875
* 'Opinion Blog': 'A1 (argumentative)': 0.66
* 'Personal Blog: 'A11 (personal)': 0.677
* 'Persuasive Article or Essay': 'A1 (argumentative)': 0.75
* 'Religious Blogs/Sermons': 'A1 (argumentative)': 0.67
* 'Reviews': 'A17 (review)': 0.72
* 'Short Story': 'A4 (fiction)': 0.80
* 'Sports Report': A8 (news)': 0.759
* 'Technical Support': 'A7 (instruction)': 0.875
* Other Informational Persuasion': 'A1 (argumentative)': 1.0
* 'Other Opinion': 'A1 (argumentative)': 0.67
* 'Other Spoken': 'A11 (personal)': 0.67


2. Categories that match, but less well:
* 'Advice': 'A7 (instruction)': 0.51
* 'Description with Intent to Sell': A12 (promotion)': 0.465, 'A17 (review)': 0.283
* 'News Report/Blog': 'A8 (news)': 0.55, 'A1 (argumentative)': 0.28
* 'Recipe': 'A7 (instruction)': 0.47, 'A11 (personal)': 0.35

3. CORE sub categories with no (appropriate) majority FTD label:
* 'Course Materials': 'A7 (instruction)': 0.30 or 'A16 (information)': 0.30
* 'Description of a Person':  'A16 (information)': 0.33, 'A17 (review)': 0.228
* 'Description of a Thing': 'A12 (promotion)': 0.37
* 'Discussion Forum': 'A1 (argumentative)': 0.37, 'A11 (personal)': 0.32
* 'FAQ about Information': 'A7 (instruction)': 0.44, 'A12 (promotion)': 0.28
* 'Historical Article': 'A16 (information)': 0.423, 'A1 (argumentative)': 0.36
* 'Information Blog': 'A1 (argumentative)': 0.29, 'A7 (instruction)': 0.21
* 'Interview': ''A17 (review)': 0.36, A1 (argumentative)': 0.18, 'A11 (personal)': 0.265
* 'Magazine Article': 'A1 (argumentative)': 0.47
* 'Poem': 'A4 (fiction)': 0.33, A11 (personal)': 0.2, 'A17 (review)': 0.27
* 'Prayer': A16 (information)': 0.67
* 'Question/Answer Forum':  'A7 (instruction)': 0.357, 'A11 (personal)': 0.23, A1 (argumentative)': 0.209
* 'Reader/Viewer Responses': 'A17 (review)': 0.3, 'A12 (promotion)': 0.3, 'A7 (instruction)': 0.2
* 'Research Article': 'A1 (argumentative)': 0.39, 'A14 (academic)': 0.376
* 'Song Lyrics': 'A11 (personal)': 0.63 (matches well, but debatable if it's appropriate category)
* 'TV/Movie Script': A11 (personal)': 0.6
* 'Technical Report': 'A1 (argumentative)': 0.375, 'A7 (instruction)': 0.25 
* 'Transcript of Video/Audio': 'A1 (argumentative)': 0.83
* 'Travel Blog': 'A11 (personal)': 0.438, A17 (review)': 0.25, 'A12 (promotion)': 0.19
* 'Other Information':  'A7 (instruction)': 0.30, 'A16 (information)': 0.185